src/fontman/ttf_font.cpp

```diff
@@ -0,0 +1,594 @@
+        m_recentPixelSize = fontSize;
+        g_loadedFaces_mutex.unlock();
+    }
+
+    t_glyphIndex = FT_Get_Char_Index(cur_font, character);
+    if(t_glyphIndex == 0)//Attempt to find a fallback
+    {
+        g_loadedFaces_mutex.lock();
+        for(TtfFont *fb_font : g_loadedFaces)
+        {
+            if(fb_font->m_recentPixelSize != fontSize)
+            {
+                error = FT_Set_Pixel_Sizes(fb_font->m_face, 0, fontSize);
+                if(error)
+                    pLogWarning("TheGlyph::loadGlyph (2) Failed to set the pixel sizes %u for the font %s: %s", fontSize, cur_font->family_name, FT_Error_String(error));
+                fb_font->m_recentPixelSize = fontSize;
+            }
+
+            t_glyphIndex = FT_Get_Char_Index(fb_font->m_face, character);
+            if(t_glyphIndex != 0)
+            {
+                // Fallback has been found!
+                cur_font = fb_font->m_face;
+                break;
+            }
+        }
+        g_loadedFaces_mutex.unlock();
+    }
+
+    FT_Int32 ftLoadFlags = FT_LOAD_RENDER;
+
+    if(!m_enableAntialias)
+        ftLoadFlags |= FT_LOAD_MONOCHROME | FT_LOAD_TARGET_MONO;
+
+    error = FT_Load_Glyph(cur_font, t_glyphIndex, ftLoadFlags);
+    if(error != 0)
+        return dummyGlyph;
+
+
+    FT_GlyphSlot glyph  = cur_font->glyph;
+    FT_Bitmap &bitmap   = glyph->bitmap;
+    uint32_t width      = bitmap.width;
+    uint32_t height     = bitmap.rows;
+    uint32_t pitch      = width * 4;
+
+    if((width == 0) || (height == 0))
+        return dummyGlyph;
+
+    SDL_assert_release(bitmap.buffer); // Buffer must NOT be null
+
+    uint8_t *image = nullptr;
+
+    try
+    {
+        size_t image_size = 4 * width * height;
+        image = new uint8_t[image_size];
+        SDL_memset(image, 0, image_size);
+    }
+    catch(const std::bad_alloc &e)
+    {
+        pLogCritical("TtfFont::TheGlyph: Out of memory: %s", e.what());
+        return dummyGlyph;
+    }
+
+    switch(bitmap.pixel_mode)
+    {
+    case FT_PIXEL_MODE_NONE:
+        break;
+
+    case FT_PIXEL_MODE_MONO:
+    {
+        uint32_t bsize = (width + 7) / 8;
+
+        if(bitmap.pitch >= 0)
+        {
+            for(uint32_t b = 0; b < bsize; ++b)
+            {
+                for(uint32_t h = 0; h < height; ++h)
+                {
+                    uint8_t block = bitmap.buffer[(static_cast<uint32_t>(bitmap.pitch) * ((height - 1) - h)) + b];
+                    uint32_t* dest = reinterpret_cast<uint32_t*>(&(image[((4 * width) * (height - 1 - h)) + (32 * b)]));
+                    for(int x = 0; x < 8 && (b * 8 + x) < width; x++)
+                    {
+                        if(block & (1 << (7 - x)))
+                            dest[x] = 0xFFFFFFFF;
+                        else
+                            dest[x] = 0;
+                    }
+                }
+            }
+        }
+        else if(bitmap.pitch >= 0)
+        {
+            for(uint32_t b = 0; b < bsize; ++b)
```

Comment:
I just realized all of these are going against memory order; probably more cache friendly to outer loop over row and inner loop over col/block.